### ISSUE TITLE
Remove deprecated twisted.python.reflect methods

### ISSUE THREAD

**Issue by twisted-trac** [twisted/twisted#8293]
|[<img alt="hawkowl's avatar" src="https://avatars.githubusercontent.com/u/3307100?s=50" width="50" height="50">](https://github.com/hawkowl)| @hawkowl reported|
|-|-|
|Trac ID|trac#8293|
|Type|enhancement|
|Created|2016-04-15 01:52:49Z|
|Branch|https://github.com/twisted/twisted/tree/removeoldreflect-8293|

These are icky and duplicate things now in the stdlib for a long time.

- funcinfo
- getcurrent
- isinst
- allYourBase
- accumulateBases

<details><summary>Searchable metadata</summary>

```
trac-id__8293 8293
type__enhancement enhancement
reporter__hawkowl hawkowl
priority__normal normal
milestone__None None
branch__branches_removeoldreflect_8293 branches/removeoldreflect-8293
branch_author__hawkowl hawkowl
status__closed closed
resolution__fixed fixed
component__core core
keywords__None None
time__1460685169967940 1460685169967940
changetime__1462877811282745 1462877811282745
version__None None
owner__hawkowl hawkowl

```
</details>

**Comment by twisted-trac** [comment]
|[<img alt="hawkowl's avatar" src="https://avatars.githubusercontent.com/u/3307100?s=50" width="50" height="50">](https://github.com/hawkowl)<a name="note_1"></a>|@hawkowl commented|
|-|-|

(In [47240]) Branching to removeoldreflect-8293.

**Comment by twisted-trac** [comment]
|[<img alt="hawkowl's avatar" src="https://avatars.githubusercontent.com/u/3307100?s=50" width="50" height="50">](https://github.com/hawkowl)<a name="note_2"></a>|@hawkowl commented|
|-|-|

Builders spun, please review.

**Comment by twisted-trac** [comment]
|[<img alt="adiroiban's avatar" src="https://avatars.githubusercontent.com/u/204609?s=50" width="50" height="50">](https://github.com/adiroiban)<a name="note_3"></a>|@adiroiban set owner to @hawkowl|
|-|-|

Hm... it looks like some of those methods are still referred in the code

```
twisted.persisted.styles._aybabtu: invalid ref to twisted.python.reflect.allYourBase
```

While checking the branch, I have removed that reference and pushed the change.

Otherwise it looks good.

Please pull my change and merge.

Thanks!

**Comment by twisted-trac** [comment]
|[<img alt="hawkowl's avatar" src="https://avatars.githubusercontent.com/u/3307100?s=50" width="50" height="50">](https://github.com/hawkowl)<a name="note_4"></a>|@hawkowl set status to `closed`|
|-|-|

In changeset e8f456aadda849a0c79454f786511b195657a8be
```
#!CommitTicketReference repository="" revision="e8f456aadda849a0c79454f786511b195657a8be"
Merge removeoldreflect-8293: Remove deprecated twisted.python.reflect methods

Author: hawkowl
Reviewer: adiroiban
Fixes: #8293
```
